### PR TITLE
Add functionality to handle vbz compression

### PR DESCRIPTION
**Pull request by aunderwo**
In order for nanopolish to be able handle the default compression for Mk1C Minions (vbz) it must be aware of where to find the plugin. Exporting the HDF5_PLUGIN_PATH found in the conda install (also used to make the biocontainer image) is the solution

See also
https://github.com/jts/nanopolish/issues/899#issuecomment-826920235

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by aunderwo** [comment]
@jonasscheid Not sure why the conda test is failing. It seems to be unrelated to my change since I am only setting an Environmental variable.

**Comment by JoseEspinosa** [comment]
Nice! 😄  I will try to check it out.

**Comment by aunderwo** [comment]
@drpatelh I'm not sure if this the nf-core idiomatic way to allow the plugin path to be specified. The default is the path in the current quay.io docker and galaxy singularity images but this may change of course when the artic bioconda package is changed. Please can you advise.

File: modules/artic/minion/main.nf [hunk 1]
```diff
@@ -43,7 +43,9 @@ process ARTIC_MINION {
         summary = ""
         model = file(medaka_model).exists() ? "--medaka-model ./$medaka_model" : "--medaka-model $medaka_model"
     }
+    def hd5_plugin_path = params.options.hd5_plugin_path ? "HDF5_PLUGIN_PATH=" + params.options.hd5_plugin_path : "HDF5_PLUGIN_PATH=/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin"
     """
+    $hd5_plugin_path \\
```
Comment (drpatelh) [inline modules/artic/minion/main.nf:48]:
```suggestion
    def hd5_plugin_path = params.options.hd5_plugin_path ? "HDF5_PLUGIN_PATH=" + params.options.hd5_plugin_path : "HDF5_PLUGIN_PATH=/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin"
    """
    HDF5_PLUGIN_PATH=/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin

```

Comment (drpatelh) [reply modules/artic/minion/main.nf:48]: I wonder whether we just hard-code it here? If it is always shipped with the container then the main issue is making sure that path exists in the container when it is updated. Can we add another test for this somehow?

Comment (drpatelh) [reply modules/artic/minion/main.nf:48]: Would the path above be different if using Conda or a Container in which case the suggestion above won't work.

**Review by JoseEspinosa — COMMENTED** [review]
Yes, actually I wondered the same and check it and seems that in the Conda environment the path is not available. I don't know if a possible solution could be to provide the path as an input variable that lets the user change it? 🤔

**Comment by aunderwo** [comment]
> Yes, actually I wondered the same and check it and seems that in the Conda environment the path is not available. I don't know if a possible solution could be to provide the path as an input variable that lets the user change it? 🤔

@JoseEspinosa 
This [commit](https://github.com/nf-core/modules/pull/1197/commits/69854132c98e2031f86646dde209a9da73778196) allows a user to change it by adding to the module.config
```
hd5_plugin_path = '/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin'
```

**Comment by aunderwo** [comment]
> > Yes, actually I wondered the same and check it and seems that in the Conda environment the path is not available. I don't know if a possible solution could be to provide the path as an input variable that lets the user change it? 🤔
> 
> @JoseEspinosa This [commit](https://github.com/nf-core/modules/pull/1197/commits/69854132c98e2031f86646dde209a9da73778196) allows a user to change it by adding to the module.config
> 
> ```
> hd5_plugin_path = '/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin'
> ```

I tried the pipeline using the `-profile conda` argument bit I need to supply the temporary work location in the config

```
            hd5_plugin_path = '/home/ubuntu/sars-cov-2_analysis/work/conda/env-1fb5f05250495b28b96a7172c968e8cc/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin'
```

**Comment by aunderwo** [comment]
@JoseEspinosa and @drpatelh. I just confirmed the issue that the viralrecon nanopolish pipeline does not work unless you add that path

Here is some data when the pipeline is run with the plugin path is specified
image
Without this the Lineage (3rd column) is all `B`, same as the reference

File: modules/artic/minion/main.nf (hunk 1 above)
Comment (drpatelh) [inline modules/artic/minion/main.nf:48]:
```suggestion
    def hd5_plugin_path = task.ext.hd5_plugin_path ? "HDF5_PLUGIN_PATH=" + task.ext.hd5_plugin_path : "HDF5_PLUGIN_PATH=/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin"
    """
    $hd5_plugin_path

```

Comment (drpatelh) [reply modules/artic/minion/main.nf:48]: Ok. I think this is the best workaround if we need to provide the path manually (and in line with the current DSL2 syntax we are using).

Comment (drpatelh) [reply modules/artic/minion/main.nf:48]: Once this is merged I will re-install it in viralrecon @aunderwo. Hoping to get a release out soon anyway.

**Review by drpatelh — APPROVED** [review]
🎉

**Comment by drpatelh** [comment]
Done https://github.com/nf-core/viralrecon/pull/243/commits/95eeceb70f79647c32d3969ec3a80318a9cb3d69. This will be on `dev` of viralrecon tomorrow to test if you like.

**Comment by JoseEspinosa** [comment]
🎉

**Comment by aunderwo** [comment]
Thanks - I'll give it a test

**Pull request merged**